### ISSUE TITLE
An important issue.

### ISSUE THREAD

**Issue by kingwmk** [ethanfetaya/NRI#17]
In the test phase, the encoder sees ground truth data that should not be seen, resulting in higher precision. May I ask for some explanation?

**Comment by tkipf** [comment]
We always (have to) condition the generative model on some initial trajectory, even at wat time — in case this is what you refer to?

**Comment by kingwmk** [comment]
the code in modules:

638          if dynamic_graph and step >= burn_in_steps:
639                # NOTE: Assumes burn_in_steps = args.timesteps
                logits = encoder(
                    data[:, :, step - burn_in_steps:step, :].contiguous(),
                    rel_rec, rel_send)

**Comment by tkipf** [comment]
Yes, we are conditioning the generative model on an initial sub-sequence (a few steps of ‘real’ data) from which our model predicts future time steps. Of course we only evaluate the predictive quality on these ‘future’ time steps and discard the conditioning. This can be seen as a ‘burn-in’ phase for the recurrent generative model, as outlined in the code that you mentioned.

**Comment by kingwmk** [comment]
Yes. However,  When employing dynamic-graph in the test step, after burnin step the encoder still use the ground-truth data as " data[:, :, step - burn_in_steps:step, :] " to infer, so it sees ground truth data that should not be seen. Is it right in such case? I mean after burnin step, should the encoder use predicted data for the reasoning?

**Comment by kingwmk** [comment]
Thank you for your kindness and  timely response. Thank you very much for your contribution to the community! I am not malicious. I just want to exchange and discuss issues and learn from each other.

**Comment by tkipf** [comment]
I see what you're saying. Yes, this is indeed using teacher forcing to re-estimate the discrete latent graph, but the decoder is only conditioned on its own past predictions + the discrete latent graph which can only communicate very little information about the (past) ground truth trajectory.
